pkg/skaffold/deploy/helm/helm_windows_test.go

```diff
@@ -0,0 +1,108 @@
+	}
+
+	testutil.Run(t, "TestSetFilesFlagInDeploy", func(t *testutil.T) {
+		expectedSetFiles := []string{
+			fmt.Sprintf(`--set-file expanded=%s\\file.yaml`, strings.ReplaceAll(home, "\\", "\\\\")),
+			`--set-file escaped.abs=C:\\Users\\foo\\escaped.abs.yaml`,
+			`--set-file escaped.relative=.\\foo\\escaped.relative.yaml`,
+
+			`--set-file escaped.spaces.abs="C:\\Users\\foo bar\\escaped.spaces.abs.yaml"`,
+			`--set-file escaped.spaces.relative=".\\foo bar\\escaped.spaces.relative.yaml"`,
+
+			`--set-file not.escaped.abs=C:\\Users\\foo\\not.escaped.abs.yaml`,
+			`--set-file not.escaped.relative=.\\foo\\not.escaped.relative.yaml`,
+
+			`--set-file not.escaped.spaces.abs="C:\\Users\\foo bar\\not.escaped.spaces.abs.yaml"`,
+			`--set-file not.escaped.spaces.relative=".\\foo bar\\not.escaped.spaces.relative.yaml"`,
+		}
+		sort.Strings(expectedSetFiles)
+		commands := testutil.
+			CmdRunWithOutput("helm version --client", version31).
+			AndRun("helm --kube-context kubecontext get all skaffold-helm --kubeconfig kubeconfig").
+			AndRun("helm --kube-context kubecontext dep build examples/test --kubeconfig kubeconfig").
+			AndRun(fmt.Sprintf(`helm --kube-context kubecontext upgrade skaffold-helm examples/test -f skaffold-overrides.yaml --set-string image=docker.io:5000/skaffold-helm:3605e7bc17cf46e53f4d81c4cbc24e5b4c495184 %s --kubeconfig kubeconfig`, strings.Join(expectedSetFiles, " "))).
+			AndRun("helm --kube-context kubecontext get all skaffold-helm --template {{.Release.Manifest}} --kubeconfig kubeconfig")
+
+		fakeWarner := &warnings.Collect{}
+		env := []string{"FOO=FOOBAR"}
+		t.Override(&warnings.Printf, fakeWarner.Warnf)
+		t.Override(&util.OSEnviron, func() []string { return env })
+		t.Override(&util.DefaultExecCommand, commands)
+		t.Override(&osExecutable, func() (string, error) { return "SKAFFOLD-BINARY", nil })
+
+		deployer, err := NewDeployer(&helmConfig{
+			configFile: "test.yaml",
+		}, nil, &latest_v1.HelmDeploy{
+			Releases: []latest_v1.HelmRelease{{
+				Name:      "skaffold-helm",
+				ChartPath: "examples/test",
+				ArtifactOverrides: map[string]string{
+					"image": "skaffold-helm",
+				},
+				Overrides: schemautil.HelmOverrides{Values: map[string]interface{}{"foo": "bar"}},
+				SetFiles: map[string]string{
+					"expanded":                    "~/file.yaml",
+					"not.escaped.relative":        `.\foo\not.escaped.relative.yaml`,
+					"not.escaped.abs":             `C:\Users\foo\not.escaped.abs.yaml`,
+					"escaped.relative":            `.\\foo\\escaped.relative.yaml`,
+					"escaped.abs":                 `C:\\Users\\foo\\escaped.abs.yaml`,
+					"escaped.spaces.relative":     `.\\foo bar\\escaped.spaces.relative.yaml`,
+					"escaped.spaces.abs":          `C:\\Users\\foo bar\\escaped.spaces.abs.yaml`,
+					"not.escaped.spaces.relative": `.\foo bar\not.escaped.spaces.relative.yaml`,
+					"not.escaped.spaces.abs":      `C:\Users\foo bar\not.escaped.spaces.abs.yaml`,
```

Comment:
Honestly, I'd rather we just have a specific test for `sanitizeFilePath` with these different permutations — including quotation tests for non-Windows — with some of the tests marked as being Windows specific.  The rest of this test is re-testing functionality already checked in `TestHelmDeploy`.  Or am I missing something?